platform/darwin/src/MGLAttributionInfo.mm

```diff
@@ -65,10 +65,20 @@ @implementation MGLAttributionInfo
     NSData *htmlData = [styledHTML dataUsingEncoding:NSUTF8StringEncoding];
 
 #if TARGET_OS_IPHONE
-    NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithData:htmlData
-                                                                                          options:options
-                                                                               documentAttributes:nil
-                                                                                            error:NULL];
+    __block NSMutableAttributedString *attributedString;
+    dispatch_block_t initialization = ^{
+            // This initializer should be called from a global or main queue. https://developer.apple.com/documentation/foundation/nsattributedstring/1524613-initwithdata
+            attributedString = [[NSMutableAttributedString alloc] initWithData:htmlData
+                                                                       options:options
+                                                            documentAttributes:nil
+                                                                         error:NULL];
+    };
+    
+    if (![[NSThread currentThread] isMainThread]) {
+        dispatch_sync(dispatch_get_main_queue(), initialization);
+    } else {
+        initialization();
+    }
 #else
     NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithHTML:htmlData
```

Comment:
Does `initWithHTML...` have the same concern?

Reply:
Nothing in the documentation: https://developer.apple.com/documentation/foundation/nsattributedstring/1535412-initwithhtml